### DISCUSSION TITLE
modulating GFPGan1.4 ?

### DISCUSSION THREAD

**Discussion by TheManBe** [chaiNNer-org/chaiNNer#2334] [General]
I just discovered chaiNNer, in the search for the best workflow for in particular facial/portrait upscale of very low quality source pictures.

Am absolutely BLOWN AWAY by a model like "4xFaceUpDAT.pth". At times,  a combination with GFPGan1.4 significantly improves the endresult even further.

I'm coming from AUTOMATIC1111 for Stable Diffusion, using the "extras" tab there to apply some of the upscalers. DAT models are not available yet, there. However, a feature that I DO use over there, is the ability to "modulate" the effect of GFPGan by tuning the "visibility" to something else than "completely ON" or"completely OFF". Most of the time, I find that setting "GFPGan visibility" to 0.75 or so, finds a better balance between "restoration" and "faithfulness".

Does chaiNNer offer the ability to do something similar?

The documentation of "Upscale Face" talks about "weight", but I don't have that, only "Output scale" (?)

![chaiNNer-Untitled-2023-11-24_23-06](https://github.com/chaiNNer-org/chaiNNer/assets/115418482/4a0577d3-7248-4211-af27-18d7f0258611)

**Comment by joeyballentine** [comment]
"weight" only applies to code former. It's a setting only available in that model.

As for that modulation setting, I'm not really sure what that does. I'll have to look into what they're doing there to know if that can currently be done with chaiNNer.

**Reply by TheManBe** [reply]
It's this setting in AUTOMATIC1111:

![Clipboard02vvv](https://github.com/chaiNNer-org/chaiNNer/assets/115418482/947e60a2-72f5-4c28-ad8c-86a3746cbd21)

**Reply by joeyballentine** [reply]
So is it just like lowering the transparency of the result over a version of the image scaled with lanczos or something?

**Reply by TheManBe** [reply]
I have absolutely no idea... I'm not a coder, lol...  It might well be something like that, though. It helps to "tune down" the strength of the facial repair.

**Reply by TheManBe** [reply]
Well, I did a little experiment with the "Z-stack" node.

Things DO seem to go in the right direction,that way. Especially with the "Min" Expression. It might be useful to have even more options and control within that "Z-Stack" node.

Thinking far ahead here, but what about some "morph" functionality with a slider?

![Z-stack_FaceUpDAT_GFPGAN](https://github.com/chaiNNer-org/chaiNNer/assets/115418482/b24ec648-f1e8-41a5-882a-f1dad73ef1f5)
